mangaki/mangaki/models.py

```diff
@@ -165,38 +165,62 @@ def get_absolute_url(self):
         return reverse('work-detail', args=[self.category.slug, str(self.id)])
 
     def retrieve_tags(self, anidb):
-        anidb_tags_list = anidb.get(self.anidb_aid).tags
-        anidb_tags = {title: int(weight) for title, weight in anidb_tags_list}
+        anidb_tags = anidb.handle_tags(anidb_aid=self.anidb_aid)
 
-        tag_work = TaggedWork.objects.filter(work=self)
-        current_tags = {tagwork.tag.title: tagwork.weight for tagwork in tag_work}
+        tag_work_list = TaggedWork.objects.filter(work=self).all()
+        values = tag_work_list.values_list('tag__title', 'tag__anidb_tag_id', 'weight')
+        current_tags = {value[0]: {"weight": value[2], "anidb_tag_id": value[1]} for value in values}
```

Comment:
For readability, if you could write this `dict` comprehension on multiple lines:
```python
current_tags = {
     value[0]: {
         "weight": value[2],
         "anidb_tag_id": value[1]
     } for value in values
}
```